kafka/source/pkg/reconciler/kafkasource.go

```diff
@@ -206,6 +219,188 @@ func checkResourcesStatus(src *v1alpha1.KafkaSource) error {
 	return nil
 }
 
+func (r *Reconciler) scaleKafkaSource(ctx context.Context, ra *v1.Deployment, src *v1alpha1.KafkaSource) {
```

Comment:
For an example of how we are doing it for our pubsub sources, you can see: https://github.com/google/knative-gcp/blob/26efd4cf70d23a7d0fde2231d941608ebf8da98b/pkg/reconciler/pubsub/pullsubscription/keda/pullsubscription.go#L63